packages/nextcloud/packages/nextcloud_test_presets/lib/src/generate_presets.dart

```diff
@@ -39,17 +42,31 @@ Future<void> generatePresets() async {
       final buffer = StringBuffer()..writeln('SERVER_VERSION=${serverRelease.dockerImageTag}');
 
       for (final a in apps) {
-        buffer
-          ..write(a.id.toUpperCase())
-          ..write('_URL=');
+        late final AppRelease appRelease;
         if (a == app) {
-          buffer.writeln(release.url);
+          appRelease = release;
         } else {
-          final release = a.findLatestCompatibleRelease(serverRelease) ??
+          appRelease = a.findLatestCompatibleRelease(serverRelease) ??
               a.findLatestCompatibleRelease(serverRelease, allowUnstable: true) ??
               a.findLatestRelease();
-          buffer.writeln(release.url);
         }
+
+        if (urlChecksums[appRelease.url] == null) {
+          final request = http.Request('GET', Uri.parse(appRelease.url));
+
+          final streamedResponse = await httpClient.send(request);
+          if (streamedResponse.statusCode != 200) {
+            throw Exception('Unable to get app, status code: ${streamedResponse.statusCode}');
+          }
+
+          final response = await http.Response.fromStream(streamedResponse);
```

Comment:
I think we should use the converter directly on the stream. I think this avoids having the entire content in memory.